File: ward_connect/lib/models/rationcard.dart
import 'dart:convert';

class Member {
  final int mid;
  final String name;
  final String relation;
  final int age;
  final String occupation;
  final int income;

  Member({
    required this.mid,
    required this.name,
    required this.relation,
    required this.age,
    required this.occupation,
    required this.income,
  });

  factory Member.fromMap(Map<String, dynamic> map) {
    return Member(
      mid: map['mid'],
      name: map['name'],
      relation: map['relation'],
      age: map['age'],
      occupation: map['occupation'],
      income: map['income'],
    );
  }
}

class Applied {
  final String rid;
  final int totalMembers;
  final List<Member> members;

  Applied({
    required this.rid,
    required this.totalMembers,
    required this.members,
  });

  Map<String, dynamic> toMap() {
    return {
      'rid': rid,
      'totalMembers': totalMembers,
      'members': members,
    };
  }

  factory Applied.fromMap(Map<String, dynamic> map) {
    List<Member> members = [];
    for (var memberData in map['members']) {
      members.add(Member.fromMap(memberData));
    }
    return Applied(
      rid: map['rid'],
      totalMembers: map['total_members'],
      members: members,
    );
  }
  String toJson() => json.encode(toMap());

  factory Applied.fromJson(String source) =>
      Applied.fromMap(json.decode(source));
}

File: ward_connect/server/models/ration_card.js
const mongoose = require("mongoose");

const memberSchema = mongoose.Schema({
    mid: {
    required: true,
    type: Number,
    trim: true,
  },
  name: {
    required: true,
    type: String,
    trim:true
  },
  relation: {
    required: true,
    type: String,
    trim:true
  },
  age:{
    required:true,
    type:Number,
    trim:true
  },
  occupation:{
    required:true,
    type:String,
    trim:true
  },
  income:{
    required:true,
    type:Number,
    trim:true
  }
});

const appliedSchema = mongoose.Schema({
    rid: {
    required: true,
    type: String,
    trim: true,
  },
  total_members: {
    required: true,
    type: Number,
    trim:true
  },
  members:[{memberSchema}]
  
});

const ration_card = mongoose.model("rationcard_members", schemeSchema);
module.exports = ration_card;
